docs/guides/09-interactivity-events.md

```diff
@@ -0,0 +1,410 @@
+
+The result should look like this:
+<div class="example-map">
+    <iframe
+        id="guides-interactivity-step-1"
+        src="/developers/carto-vl/examples/maps/guides/interactivity/step-1.html"
+        width="100%"
+        height="500"
+        frameBorder="0">
+    </iframe>
+</div>
+
+
+### Layer events
+Once you have your basemap, you can start to work with your layers. And all [carto.Layer](/developers/carto-vl/reference/#cartolayer) objects have two events to listen to: `loaded` and `updated`.
+
+The use of the `loaded` event is pretty common, due to the fact that in most cases you need to load the data from an external server and that can take some time.
+
+Add this to your code to create a layer as usual, and add it to your map:
+```js
+carto.setDefaultAuth({ username: 'cartovl', apiKey: 'default_public' });
+const source = new carto.source.Dataset('populated_places');
+const viz = new carto.Viz();
+const layer = new carto.Layer('Cities', source, viz);
+layer.addTo(map);
+```
+
+Now add a `loaded` listener to the previous layer:
+```js
+layer.on('loaded', () => {
+    console.log('Cities layer has been loaded!');
+});
+```
+> This event could be useful for example to display some kind of loading animation over your map and then hiding it after the layer has loaded (see for example this [visualization](/developers/carto-vl/examples/maps/advanced/landing-page/hurricane-harvey.html)). Notice how the name of the event is `loaded`, not `load`.
+
+If you were using several layers, you could also have a single function to handle them all. For this case, **on** and **off** are available at `carto` namespace, expecting a list of layers like in this code: `carto.on('loaded', [layer1, layer2], () => { console.log('All layers have loaded'); })`.
+
+Regarding to the `updated` event, it can be useful for the cases where the layer's viz changes, for example when you are building an animation. See the [Playing with animations guide](/developers/carto-vl/guides/playing-with-animations).
+
+If you check your work now, it should look like this:
+<div class="example-map">
+    <iframe
+        id="guides-interactivity-step-2"
+        src="/developers/carto-vl/examples/maps/guides/interactivity/step-2.html"
+        width="100%"
+        height="500"
+        frameBorder="0">
+    </iframe>
+</div>
+
+
+### Using dynamic variables
+*Variables* are a way to store and reuse expressions, and that can definitively help you when adding interactions to your visualization, so let's practice a bit with them.
+
+#### Variables without properties
+First, you are going to add a variable whose value depends solely on the current map extent. Replace your current `const viz = new carto.Viz();`, with this code that grabs the current displayed features using the *String API*:
+```js
+const viz = new carto.Viz(`
+    @currentFeatures: viewportFeatures()
```

Comment:
I think this is not valid, `viewportFeatures()` requires properties as the paramters

Reply:
@davidmanzanares it seems it is working fine in the examples for this guide. 
I took it from /debug/viewport.html where it doesn't use parameters either.

Reply:
When using viewportFeatures without specific properties (like in debug/viewport.html) `viz.variables.currentFeatures.value` holds an array of ViewporFeature, which have cartodb_id

![image](https://user-images.githubusercontent.com/458196/46618633-56c5dd80-cb20-11e8-9af0-da1d56205f2e.png)

Reply:
If we don't want this behaviour 🔝, we should talk and fix the guide and/or the jsdoc

Reply:
I have replaced unintended use of `viewportFeatures` to use `viewportCount`, which is more suitable to the task, specially now it does not required any property as input